File: solutions/python/algorithm/linkedlist/Exercise2.py
from structure.dynamic.DynamicLinkedListADT import DynamicLinkedListADT
from util.LinkedListUtil import print_list


def remove_last(linked_list, n):
    total_size = linked_list.size()

    if n >= total_size:
        for i in range(total_size - 1, -1, -1):
            linked_list.remove(i)
    else:
        for _ in range(n):
            linked_list.remove(linked_list.size() - 1)


# 🧪 Ejemplo de uso
lista = DynamicLinkedListADT()
lista.add(10)
lista.add(20)
lista.add(30)
lista.add(40)

print("Lista Original:")
print_list(lista)

remove_last(lista, 2)

print("Lista sin elementos:")
print_list(lista)

File: solutions/javascript/src/algorithm/linkedlist/Exercise2.js
import DynamicLinkedListADT from "../../structure/dynamic/DynamicLinkedListADT.js";


function printList(linkedList) {
  const elements = [];
  for (let i = 0; i < linkedList.size(); i++) {
    elements.push(linkedList.get(i));
  }
  console.log(elements);
}

function removeLast(linkedList, n) {
    const totalSize = linkedList.size();

    if (n >= totalSize) {
        for (let i = totalSize - 1; i >= 0; i--) {
            linkedList.remove(i);
        }
    } else {
        for (let i = 0; i < n; i++) {
            linkedList.remove(linkedList.size() - 1);
        }
    }
}


const lista = new DynamicLinkedListADT();
lista.add(10);
lista.add(20);
lista.add(30);

console.log("Lista original");
printList(lista); // Salida: [10, 20, 30]

removeLast(lista,2);
console.log("Lista con eliminados");
printList(lista); // Salida: [10]